exporter/clickhouseexporter/exporter.go

```diff
@@ -73,27 +74,30 @@ func (e *clickhouseExporter) pushLogsData(ctx context.Context, ld plog.Logs) err
 		defer func() {
 			_ = statement.Close()
 		}()
+		var serviceName string
 		for i := 0; i < ld.ResourceLogs().Len(); i++ {
 			logs := ld.ResourceLogs().At(i)
 			res := logs.Resource()
-			resourceKeys, resourceValues := attributesToSlice(res.Attributes())
+			resAttr := attributesToMap(res.Attributes())
+			if v, ok := res.Attributes().Get(conventions.AttributeServiceName); ok {
```

Comment:
Do you want to keep `service.name` duplicated in the attributes?

Reply:
yes, just copy service.name to separate column for primary index, not modify the original attributes.